File: Remove Duplicates from Sorted Array/Remove Duplicates from Sorted Array.java
public class Solution {
    public int removeDuplicates(int[] A) {
        int len = A.length;
        if(len == 0 || len == 1){
            return len;
        }
        
        int firstIndex = 0;
        int secondIndex = 1;
        while(secondIndex < len){
            if(A[firstIndex] == A[secondIndex]){
                // Do nothing
            }
            else{
                ++firstIndex;
                A[firstIndex] = A[secondIndex];
            }
            ++secondIndex;
            
        }
        return firstIndex + 1;
    }
}

File: Remove Duplicates from Sorted Array/Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        if(!n || (n == 1))  return n;
        
        int mark = 0;
        for(unsigned int i = 1; i < n; ++i){
            if(A[mark] != A[i]){
                ++mark;
                A[mark] = A[i];
            }
        }
        
        return mark + 1;
    }
};
